diff -pruN zstd-1.4.4/contrib/pzstd/Makefile zstd-1.4.4.1/contrib/pzstd/Makefile
--- zstd-1.4.4/contrib/pzstd/Makefile	2019-11-05 02:54:32.000000000 +0900
+++ zstd-1.4.4.1/contrib/pzstd/Makefile	2020-03-03 10:16:41.217252562 +0900
@@ -171,7 +171,9 @@ roundtripcheck: roundtrip check
 	$(TESTPROG) ./test/RoundTripTest$(EXT) $(TESTFLAGS)
 
 # Build the main binary
-pzstd$(EXT): main.o $(PROGDIR)/util.o Options.o Pzstd.o SkippableFrame.o $(ZSTDDIR)/libzstd.a
+
+pzstd$(EXT): LIBS += -lzstd -L../../lib
+pzstd$(EXT): main.o $(PROGDIR)/util.o Options.o Pzstd.o SkippableFrame.o
 	$(LD_COMMAND)
 
 # Target that depends on all the tests
@@ -186,8 +188,11 @@ roundtrip: test/RoundTripTest$(EXT)
 
 # Use the static library that zstd builds for simplicity and
 # so we get the compiler options correct
+
 $(ZSTDDIR)/libzstd.a: $(ZSTD_FILES)
-	CFLAGS="$(ALL_CFLAGS)" LDFLAGS="$(ALL_LDFLAGS)" $(MAKE) -C $(ZSTDDIR) libzstd.a
+	LDFLAGS=-lzstd
+#$(ZSTDDIR)/libzstd.a: $(ZSTD_FILES)
+#	CFLAGS="$(ALL_CFLAGS)" LDFLAGS="$(ALL_LDFLAGS)" $(MAKE) -C $(ZSTDDIR) libzstd.a
 
 # Rules to build the tests
 test/RoundTripTest$(EXT): test/RoundTripTest.o $(PROGDIR)/datagen.o \
diff -pruN zstd-1.4.4/programs/Makefile zstd-1.4.4.1/programs/Makefile
--- zstd-1.4.4/programs/Makefile	2019-11-05 02:54:32.000000000 +0900
+++ zstd-1.4.4.1/programs/Makefile	2020-03-03 21:02:12.071987249 +0900
@@ -207,7 +207,6 @@ zstd-noxz : LZMALD  :=
 zstd-noxz : LZMA_MSG := - xz/lzma support is disabled
 zstd-noxz : zstd
 
-
 zstd-pgo :
 	$(MAKE) clean
 	$(MAKE) zstd MOREFLAGS=-fprofile-generate
@@ -225,6 +224,9 @@ zstd-small: CFLAGS = -Os -s
 zstd-frugal zstd-small: $(ZSTD_FILES) zstdcli.c util.c timefn.c fileio.c
 	$(CC) $(FLAGS) -DZSTD_NOBENCH -DZSTD_NODICT $^ -o $@$(EXT)
 
+zstd-mini: zstdcli.c util.c timefn.c fileio.c
+	$(CC) $(FLAGS) -DZSTD_NOBENCH -DZSTD_NODICT $^ -lzstd -L../lib -Wl,-rpath=../lib -o $@$(EXT)
+
 zstd-decompress: $(ZSTDCOMMON_FILES) $(ZSTDDECOMP_FILES) zstdcli.c util.c timefn.c fileio.c
 	$(CC) $(FLAGS) -DZSTD_NOBENCH -DZSTD_NODICT -DZSTD_NOCOMPRESS $^ -o $@$(EXT)
 
